src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,1113 @@
+        w4 = rho_p * v3
+        w5 = -rho_p
+        rho_p_plus += rho_p
+
+        set_component!(entropy, k, w1, w2, w3, w4, w5, equations)
+    end
+
+    # Additional non-conservative variables
+    entropy[1] = rho_p_plus * B1
+    entropy[2] = rho_p_plus * B2
+    entropy[3] = rho_p_plus * B3
+    entropy[end] = rho_p_plus * psi
+
+    return SVector(entropy)
+end
+
+# Convert primitive to conservative variables
+@inline function prim2cons(prim, equations::IdealGlmMhdMultiIonEquations2D)
+    @unpack gammas = equations
+    B1, B2, B3 = magnetic_field(prim, equations)
+    psi = divergence_cleaning_field(prim, equations)
+
+    cons = zero(MVector{nvariables(equations), eltype(prim)})
+    cons[1] = B1
+    cons[2] = B2
+    cons[3] = B3
+    for k in eachcomponent(equations)
+        rho, v1, v2, v3, p = get_component(k, prim, equations)
+        rho_v1 = rho * v1
+        rho_v2 = rho * v2
+        rho_v3 = rho * v3
+
+        rho_e = p / (gammas[k] - 1.0) +
+                0.5f0 * (rho_v1 * v1 + rho_v2 * v2 + rho_v3 * v3) +
+                0.5f0 * (B1^2 + B2^2 + B3^2) +
+                0.5f0 * psi^2
+
+        set_component!(cons, k, rho, rho_v1, rho_v2, rho_v3, rho_e, equations)
+    end
+    cons[end] = psi
+
+    return SVector(cons)
+end
+
+# Compute the fastest wave speed for ideal multi-ion GLM-MHD equations: c_f, the fast 
+# magnetoacoustic eigenvalue. This routine computes the fast magnetosonic speed for each ion
+# species using the single-fluid MHD expressions and approximates the multi-ion c_f as 
+# the maximum of these individual magnetosonic speeds.
+@inline function calc_fast_wavespeed(cons, orientation::Integer,
+                                     equations::IdealGlmMhdMultiIonEquations2D)
+    B1, B2, B3 = magnetic_field(cons, equations)
+    psi = divergence_cleaning_field(cons, equations)
+
+    c_f = zero(real(equations))
+    for k in eachcomponent(equations)
+        rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, cons, equations)
+
+        v1 = rho_v1 / rho
+        v2 = rho_v2 / rho
+        v3 = rho_v3 / rho
+        v_mag = sqrt(v1^2 + v2^2 + v3^2)
+        gamma = equations.gammas[k]
+        p = (gamma - 1) *
+            (rho_e - 0.5f0 * rho * v_mag^2 - 0.5f0 * (B1^2 + B2^2 + B3^2) -
+             0.5f0 * psi^2)
+        a_square = gamma * p / rho
+        sqrt_rho = sqrt(rho)
+
+        b1 = B1 / sqrt_rho
+        b2 = B2 / sqrt_rho
+        b3 = B3 / sqrt_rho
+        b_square = b1^2 + b2^2 + b3^2
+
+        if orientation == 1
+            c_f = max(c_f,
+                      sqrt(0.5f0 * (a_square + b_square) +
+                           0.5f0 *
+                           sqrt((a_square + b_square)^2 - 4.0f0 * a_square * b1^2)))
+        else #if orientation == 2
+            c_f = max(c_f,
+                      sqrt(0.5f0 * (a_square + b_square) +
+                           0.5f0 *
+                           sqrt((a_square + b_square)^2 - 4.0f0 * a_square * b2^2)))
```

Comment:
```suggestion
                           sqrt((a_square + b_square)^2 - 4 * a_square * b1^2)))
        else #if orientation == 2
            c_f = max(c_f,
                      sqrt(0.5f0 * (a_square + b_square) +
                           0.5f0 *
                           sqrt((a_square + b_square)^2 - 4 * a_square * b2^2)))
```
As above

Reply:
I replaced all occurrences of `2.0f0` and `4.0f0` in feac4b897957410790dfbfce43175aed4f0d53dc.